src/API/Site/Controllers/MerchantCenter/PhoneVerificationController.php

```diff
@@ -0,0 +1,165 @@
+		$verification_method = [
+			'description'       => __( 'Method used to verify the phone number.', 'google-listings-and-ads' ),
+			'enum'              => [
+				PhoneVerification::VERIFICATION_METHOD_SMS,
+				PhoneVerification::VERIFICATION_METHOD_PHONE_CALL,
+			],
+			'required'          => true,
+			'type'              => 'string',
+			'validate_callback' => 'rest_validate_request_arg',
+		];
+
+		$this->register_route(
+			'/mc/phone-verification/request',
+			[
+				[
+					'methods'             => TransportMethods::CREATABLE,
+					'callback'            => $this->get_request_phone_verification_callback(),
+					'permission_callback' => $this->get_permission_callback(),
+					'args'                => [
+						'phone_region_code'   => [
+							'description'       => __( 'Two-letter country code (ISO 3166-1 alpha-2) for the phone number.', 'google-listings-and-ads' ),
+							'required'          => true,
+							'type'              => 'string',
+							'validate_callback' => 'rest_validate_request_arg',
+						],
+						'phone_number'        => [
+							'description'       => __( 'The phone number to verify.', 'google-listings-and-ads' ),
+							'required'          => true,
+							'type'              => 'string',
+							'validate_callback' => 'rest_validate_request_arg',
+						],
+						'verification_method' => $verification_method,
+					],
+				],
+			]
+		);
+
+		$this->register_route(
+			'/mc/phone-verification/verify',
+			[
+				[
+					'methods'             => TransportMethods::CREATABLE,
+					'callback'            => $this->get_verify_phone_callback(),
+					'permission_callback' => $this->get_permission_callback(),
+					'args'                => [
+						'verification_id'     => [
+							'description'       => __( 'The verification ID returned by the /request call.', 'google-listings-and-ads' ),
+							'required'          => true,
+							'type'              => 'string',
+							'validate_callback' => 'rest_validate_request_arg',
+						],
+						'verification_code'   => [
+							'description'       => __( 'The verification code that was sent to the phone number for validation.', 'google-listings-and-ads' ),
+							'required'          => true,
+							'type'              => 'string',
+							'validate_callback' => 'rest_validate_request_arg',
+						],
+						'verification_method' => $verification_method,
+					],
+				],
+			]
+		);
+	}
+
+	/**
+	 * Get callback for requesting phone verification endpoint.
+	 *
+	 * @return callable
+	 */
+	protected function get_request_phone_verification_callback(): callable {
+		return function( Request $request ) {
+			try {
+				$verification_id = $this->phone_verification->request_phone_verification(
+					$request->get_param( 'phone_region_code' ),
+					new PhoneNumber( $request->get_param( 'phone_number' ) ),
+					$request->get_param( 'verification_method' ),
+				);
+				return [
+					'verification_id' => $verification_id,
+				];
```

Comment:
:question: 
As @eason9487 mentioned at https://github.com/woocommerce/google-listings-and-ads/issues/962#issuecomment-911157548  it's kinda unclear what to do as there is no verification status info.

Here is the only moment we are sure, the phone is verified. So, If I understood @eason9487 's comment that's the moment when we should save the number.

Personally, I think it's the only moment when we could be sure, the user didn't manipulate the browser state, and save the unverified number.

However, it would make REST API a bit messy, plus our UI's copy doesn't suggest we are saving on "Verify" and "Save changes" saves only address info. Alternatively, we could leave it to the frontend to store the verified state (opening it for manipulations). That would make it easier to implement the UI suggested by the copy form Figma, and make it prepared for the new Google API (which I guess it's not likely to come soon)


So I thin, we need to align our visions and those two PRs (https://github.com/woocommerce/google-listings-and-ads/pull/1006 and https://github.com/woocommerce/google-listings-and-ads/pull/1019/files

Reply:
I think the conversation about this is still on going on this issue: https://github.com/woocommerce/google-listings-and-ads/issues/962. It probably makes sense to continue discussing there.

My understanding is that we will not be showing a `verification` status, rather we will be showing when a number is `unverified` by checking for `accountstatus` issues: https://github.com/woocommerce/google-listings-and-ads/issues/962#issuecomment-916672715

Reply:
Thanks.
My doubt here was mainly about. Whether the phone verification request here only verifies the number or does it also save the number. According to what @eason9487 wrote in https://github.com/woocommerce/google-listings-and-ads/pull/1019#discussion_r714460853 it's handled and saved on Google end.